flink-python/src/main/java/org/apache/flink/python/PythonDependencyManager.java

```diff
@@ -0,0 +1,183 @@
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is used to parse the information of python dependency and environment management
+ * stored in GlobalJobParameters.
+ * The parse result will be used to create PythonEnvironmentManager.
+ */
+@Internal
+public class PythonDependencyManager {
+
+	public static final String PYTHON_FILE_MAP = "PYTHON_FILE_MAP";
+	public static final String PYTHON_REQUIREMENTS = "PYTHON_REQUIREMENTS";
+	public static final String PYTHON_REQUIREMENTS_DIR = "PYTHON_REQUIREMENTS_DIR";
+	public static final String PYTHON_ARCHIVES_MAP = "PYTHON_ARCHIVES_MAP";
+	public static final String PYTHON_EXEC = "PYTHON_EXEC";
+
+	private Map<String, String> filesInPythonPath;
+	private String requirementsFilePath;
+	private String requirementsDirPath;
+	private String pythonExec;
+	private Map<String, String> archives;
+
+	/**
+	 * @param filesInPythonPath Local path and origin file name of user python files uploaded by
+	 *                          TableEnvironment#add_python_file() or command option "-pyfs".
+	 *                          Key is local absolute path and value is origin file name.
+	 * @param requirementsFilePath The file path of requirements.txt file specified by
+	 *                             TableEnvironment#set_python_requirements() or command option "-pyreq".
+	 * @param requirementsDirPath The path of the directory uploaded by TableEnvironment#set_python_requirements()
+	 *                            or command option "-pyreq". It is used to support installing python packages offline.
+	 *                            If exists it should contain all the packages listed in requirements.txt file.
+	 * @param pythonExec The path of python interpreter (e.g. /usr/local/bin/python).
+	 *                   It can be specified by TableConfig#set_python_executable() or command option "-pyexec".
+	 * @param archives Local path and target directory name of user python archives uploaded by
+	 *                 TableEnvironment#add_python_archive() or command option "-pyarch".
+	 *                 Key is local absolute path and value is target directory name.
+	 */
+	public PythonDependencyManager(
+		Map<String, String> filesInPythonPath,
+		String requirementsFilePath,
+		String requirementsDirPath,
+		String pythonExec,
+		Map<String, String> archives) {
+		this.filesInPythonPath = filesInPythonPath;
+		this.requirementsFilePath = requirementsFilePath;
+		this.requirementsDirPath = requirementsDirPath;
+		this.pythonExec = pythonExec;
+		this.archives = archives;
+	}
+
+	/**
+	 * @return Local path and origin file name of user python files uploaded by
+	 * 	       TableEnvironment#add_python_file() or command option "-pyfs".
+	 * 	       Key is local absolute path and value is origin file name.
+	 */
+	public Map<String, String> getFilesInPythonPath() {
```

Comment:
Always use Optional to return nullable values. See more details in [Code Style and Quality Guide](https://flink.apache.org/contributing/code-style-and-quality-java.html)